data_safe_haven/infrastructure/programs/sre/networking.py

```diff
@@ -399,73 +402,14 @@ def __init__(
             opts=child_opts,
             tags=child_tags,
         )
-        nsg_data_configuration = network.NetworkSecurityGroup(
+        self.nsg_data_configuration = network.NetworkSecurityGroup(
             f"{self._name}_nsg_data_configuration",
             location=props.location,
             network_security_group_name=f"{stack_name}-nsg-data-configuration",
             resource_group_name=props.resource_group_name,
             security_rules=[
```

Comment:
I think we'd be better off defining security rules in-line above, for example with something like:

```python
security_rules = [
    network.SecurityRuleArgs(...),
    ...
    network.SecurityRuleArgs(...),
]
if use_software_repositories:
  security_rules.extend([
    network.SecurityRuleArgs(...),
    ...
    network.SecurityRuleArgs(...),
  ])
```